ecommerce/extensions/payment/processors/fomopay.py

```diff
@@ -0,0 +1,204 @@
+    """FOMO Pay payment processor (September 2019).
+
+    For reference, see
+    https://developers.fomopay.com/docs/api/wechat/qr
+    """
+
+    NAME = 'fomopay'
+
+    def __init__(self, site):
+        """
+        Constructs a new instance of the FOMO Pay processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor
+        """
+
+        super(Fomopay, self).__init__(site)
+
+        configuration = self.configuration
+        self.merchant = configuration['api_username']
+        self.timeout = configuration['timeout']
+        self.callback_url = configuration['callback_url']
+        self.type = configuration['type']
+        self.api_key = configuration['api_key']
+        self.api_url = configuration['api_url']
+
+    def get_transaction_parameters(self, basket, request=None, use_client_side_checkout=False, **kwargs):
+        """
+        Generate a dictionary of signed parameters FOMO Pay requires to complete a transaction.
+
+        Arguments:
+            basket (Basket): The basket of products being purchased.
+
+        Returns:
+            dict: Payment processor-specific parameters required to complete a transaction.
+        """
+
+        parameters = {
+            'merchant': self.merchant,
+            'price': str(basket.total_incl_tax),
+            'description': self.get_basket_description(basket),
+            'transaction': basket.order_number,
+            'callback_url': self.callback_url,
+            'currency_code': basket.currency.lower(),
+            'type': self.type,
+            'timeout': self.timeout,
+            'nonce': uuid.uuid4().hex,
+        }
+
+        parameters['signature'] = self._generate_signature(parameters)
+        parameters['payment_page_url'] = reverse('fomopay:payment')
+        parameters['api_url'] = self.api_url
+        return parameters
+
+    def handle_processor_response(self, response, basket=None):
+        """
+        Handle a response (i.e., "merchant notification") from FOMO Pay.
+
+        This method does the following:
+            1. Verify the validity of the response.
+            2. Create PaymentEvents and Sources for successful payments.
+
+        Arguments:
+            response (dict): Dictionary of parameters received from the payment processor.
+
+        Keyword Arguments:
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Raises:
+            TransactionDeclined: Indicates the payment was declined by the processor.
+            GatewayError: Indicates a general error on the part of the processor.
+        """
+
+        # Validate the signature
+        if not self.is_signature_valid(response):
+            raise InvalidSignatureError
+
+        # Raise an exception for payments that were not accepted. Consuming code should be responsible for handling
+        # and logging the exception.
+        decision = response['result']
+        if decision != '0':
```

Comment:
We could change '0' to a class constant called 
```suggestion
        if decision != :
```

Reply:
You're right, done.